src/js-utils/Glennsl__BsBenchmarkJs__JsUtils.re

```diff
@@ -0,0 +1,18 @@
+module Obj = {
+
+  let immutableUpdate = (~props, obj) => Js.Obj.(()->empty->assign(obj)->assign(props));
+  
+  let shallowCopy = [%bs.raw {|function(obj) {
+      var _obj = {};
+      var keys = Object.keys(obj);
+      var length = keys.length;
+      for (var i = 0; i < length; ++i) {
+        var key = keys[i];
+        _obj[key] = obj[key];
+      };
+      return _obj;
+    }|}];
```

Comment:
This looks like it's just `Js.Obj.assign` partially applied with `Js.Obj.empty`.

Reply:
For this, I tried to keep it in line with the ES5 specification. JS `Object.assign` is from the ES6 specification, and fails to work in any version of Internet Explorer. That may be okay, especially since BuckleScript seems to be heading toward ES6 conventions anyway. But I wanted to at least try to support old browsers where possible.

Reply:
Ah, I didn't realize it was ES6. I think as long as it's offered by BuckleScript, which has historically been very conservative, it should be fine.

Reply:
It may be worth including a polyfill library in our dependencies, just to allow this to run on most platforms. Idk...

Reply:
Since this is a benchmark library it won't be used by end-users, and I don't see that it makes business-sense to optimize for non-ES6 platforms anymore either. So it doesn't seem worth the bother. But I'm open to reconsidering if there's a use-case for it.

Reply:
Although if you're going to maintain this yourself, you're of course free to do so in whatever way you want 🙂

Reply:
>it won't be used by end-users

Right. I was just thinking about how, for some dev teams, testing the performance of some algorithm on older platforms might help them develop better software if they have to support those platforms. Some of us still have to support IE6, even if Microsoft won't. XD

Reply:
Support, yes, but I doubt the benefit of optimizing for it. Some might even explicitly _not_ want to optimize for it to encourage upgrading, as the maintenance burden of having to support ancient platforms costs a lot already.

Reply:
That's a good point. And besides, it's not that hard for people to just include polyfill libraries themselves.